File: StrategyPattern_C++/ConcreteCalculateInterest2.cpp
#include "ConcreteCalculateInterest2.h"
#include "Account.h"


ConcreteCalculateInterest2::ConcreteCalculateInterest2()
{
}


ConcreteCalculateInterest2::~ConcreteCalculateInterest2()
{
}

int ConcreteCalculateInterest2::calculateInterest(Account* acc)
{
	int balance = acc->getBalance();
	int interest;
	if (balance >= 0)
	{
		interest = (balance * 103) / 100;
	}
	else
	{
		interest = (balance * 107) / 100;
	}
	return interest;
}

File: Strategy/Java/bank/ConcreteCalculateInterest2.java
package bank;

public class ConcreteCalculateInterest2 implements CalculateInterestStrategy
{	
	public int calculateInterest(Account acc)
	{
		int balance = acc.getBalance();
		if(balance >= 0)
		{
			int interest = (balance * 103) / 100;
			return interest;
		}
		else
		{
			int interest = (balance * 107) / 100;
			return interest;
		}
	}
}
